proposals/0000-unicode-for-string-processing.md

```diff
@@ -0,0 +1,1321 @@
+}
+
+public struct RegexRepetitionBehavior {
+  /// Match as much of the input string as possible, backtracking when
+  /// necessary.
+  public static var eager: RegexRepetitionBehavior { get }
+
+  /// Match as little of the input string as possible, expanding the matched
+  /// region as necessary to complete a match.
+  public static var reluctant: RegexRepetitionBehavior { get }
+
+  /// Match as much of the input string as possible, performing no backtracking.
+  public static var possessive: RegexRepetitionBehavior { get }
+}
+```
+
+The `RegexBuilder` quantifier APIs include a `nil`-defaulted optional `behavior` parameter. When you pass `nil`, the quantifier uses the default behavior as set by this option. If an explicit behavior is passed, that behavior is used regardless of the default.
+
+```swift
+// Example `OneOrMore` initializer
+extension OneOrMore {
+    public init<W, C0, Component: RegexComponent>(
+    _ behavior: RegexRepetitionBehavior? = nil,
+    @RegexComponentBuilder _ component: () -> Component
+  ) where Output == (Substring, C0), Component.Output == (W, C0)
+}
+```
+
+#### Single line mode (`.` matches newlines)
+
+The "any" metacharacter (`.`) matches any character in a string *except* newlines by default. With the `s` option enabled, `.` matches any character including newlines.
+
+```swift
+let str = """
+    <<This string
+    uses double-angle-brackets
+    to group text.>>
+    """
+    
+str.firstMatch(of: /<<.+>>/)        // nil
+str.firstMatch(of: /(?s)<<.+>>/)    // "This string\nuses double-angle-brackets\nto group text."
+```
+
+This option applies only to `.` used in regex syntax and does _not_ affect the behavior of `CharacterClass.any`, which always matches any character or Unicode scalar. To get the default `.` behavior when using `RegexBuilder` syntax, use `CharacterClass.anyNonNewline`.
+
+**Regex syntax:** `(?s)...` or `(?s...)`
+
+**`RegexBuilder` API:**
+
+```swift
+extension RegexComponent {
+  /// Returns a regular expression where the start and end of input
+  /// anchors (`^` and `$`) also match against the start and end of a line.
+  public func dotMatchesNewlines(_ dotMatchesNewlines: Bool = true) -> Regex<RegexOutput>
+}
+```
+
+#### Multiline mode
+
+By default, the start and end anchors (`^` and `$`) match only the beginning and end of a string. With the `m` or the option, they also match the beginning and end of each line.
+
+```swift
+let str = """
+    abc
+    def
+    ghi
+    """
+	
+str.firstMatch(of: /^abc/)          // "abc"
+str.firstMatch(of: /^abc$/)         // nil
+str.firstMatch(of: /(?m)^abc$/)     // "abc"
+	
+str.firstMatch(of: /^def/)          // nil
+str.firstMatch(of: /(?m)^def$/)     // "def"
+```
+
+This option applies only to anchors used in a regex literal. The anchors defined in `RegexBuilder` are specific about matching at the start/end of the input or the line, and therefore are not affected by this option.
```

Comment:
Seems again like we also want equivalents of `$` and `^` as API. Those can just be a `if` check on options before forwarding onwards, and that's how the documentation can read.

Reply:
As with `.`, we don't want or need to emulate the unclear behavior of `^` and `$` in the `RegexBuilder` API. When writing those symbols in a regex, they always have a specific meaning; using `RegexBuilder` API we can give those specific meanings specific names.

The proposal mislabels the options as being a part of the `RegexBuilder` module; they're really part of `_StringProcessing`. Sorry about that! I'll update those labels.